src/Controller/ModelCollector.php

```diff
@@ -412,16 +466,14 @@ private function searchParentOfInParentedMode(ModelInterface $model)
      * @param ModelInterface $model The model to search the parent of.
      *
      * @return ModelInterface|null
-     *
-     * @throws DcGeneralInvalidArgumentException When a root model has been passed.
      */
     private function searchParentOfInHierarchical(ModelInterface $model)
     {
-        if ($this->isRootModel($model)) {
-            throw new DcGeneralInvalidArgumentException('Invalid condition, root models can not have parents!');
+        $condition = $this->relationships->getChildCondition($this->rootProviderName, $this->defaultProviderName);
```

Comment:
This will only work correctly if `$model` is a direct child of the `root provider`.

This will fail on `tl_page` => `tl_article` => `tl_content` with:
- `defaultProviderName: tl_content`
- `rootProviderName: tl_page`
as `tl_article` would be the correct parent then.

The `searchParentOfIn()` method handles these conditions correctly in "top to bottom" manner, we need them here implemented as "bottom to top" and if it did not work out, we need to fall through to the previous implementation.

Reply:
@xtra That is right. I think we rework this, if the missing tree mode implement it #481.